packages/manager/modules/pci/src/projects/project/private-networks/add/add.service.js

```diff
@@ -1,61 +1,126 @@
+      })
+      .then(({ data: { id } }) =>
+        this.checkPrivateNetworkCreationStatus(projectId, id),
       )
-      .then((network) => this.createSubnets(serviceName, network.id, subnets));
-  }
-
-  createSubnets(serviceName, networkId, subnets) {
-    return Promise.all(
-      subnets.map(
-        (subnet) =>
-          this.OvhApiCloudProjectNetworkPrivateSubnet.v6().save(
-            {
-              serviceName,
-              networkId,
-            },
-            subnet,
-          ).$promise,
-      ),
-    );
+      .then(({ resourceId }) => {
+        this.Poller.kill({ namespace: 'private-network-creation' });
+        return this.getCreatedSubnet(projectId, region, resourceId);
+      });
   }
 
-  checkPrivateNetworkCreationStatus(serviceName, networkId) {
-    this.OvhApiCloudProjectNetworkPrivate.v6().resetCache();
+  checkPrivateNetworkCreationStatus(projectId, operationId) {
     return this.Poller.poll(
-      `/cloud/project/${serviceName}/network/private/${networkId}`,
+      `/cloud/project/${projectId}/operation/${operationId}`,
       {},
       {
         method: 'get',
         successRule: {
-          status: NETWORK_ACTIVE_STATUS,
+          status: 'completed',
         },
+        namespace: 'private-network-creation',
       },
     );
   }
 
+  getCreatedSubnet(projectId, region, networkId) {
+    return this.$http
+      .get(
+        `/cloud/project/${projectId}/region/${region}/network/${networkId}/subnet`,
+      )
+      .then(({ data }) => data);
+  }
+
   static generateNetworkAddress(vlanId) {
     return DEFAULT_IP.replace('{vlanId}', vlanId % 255);
   }
+
+  getAllRegions(projectId) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/region`)
+      .then(({ data }) => data);
+  }
+
+  getRegionInfo(projectId, region) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/region/${region}`)
+      .then(({ data }) => data);
+  }
+
+  getGateways(projectId, regionName) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/region/${regionName}/gateway`)
+      .then(({ data }) => data);
+  }
+
+  enableSnatOnGateway(projectId, region, gatewayId) {
+    return this.$http
+      .post(
+        `/cloud/project/${projectId}/region/${region}/gateway/${gatewayId}/expose`,
+      )
+      .then(({ data }) => data);
+  }
+
+  getSmallestGatewayInfo(ovhSubsidiary) {
+    return this.$http
+      .get('/order/catalog/public/cloud', {
+        params: {
+          ovhSubsidiary,
+          productName: 'cloud',
+        },
+      })
+      .then(({ data }) => {
+        // plans are sorted in descending order, for smallest plan pick last 2 entries
+        const gatewayProducts = data.addons
+          .filter((addon) => addon.product.startsWith('publiccloud-gateway'))
+          .slice(-2);
+        const [monthlyPriceObj] = gatewayProducts.find(({ planCode }) =>
+          planCode.includes('month'),
+        )?.pricings;
```

Comment:
Maybe we should just rely on some min function instead, there are high chances that plans order randomly changes

Reply:
Done https://github.com/ovh/manager/commit/684a600dc5c9158052884ba1e5fb6499e17368e2